Restore bug

dmitrybublik opened the pull request.

dmitrybublik committed:
Fixed snapshot structure.

dmitrybublik committed:
Fixed tests.

dmitrybublik committed:
Moved new code to new CompleSnapshot class.

dmitrybublik committed:
ComplexSnapshot: small fixes.

dmitrybublik committed:
Added EditablePropertyPropxyAttribute. Fixed tests.

godrose committed:
#4 Minor

godrose committed:
Merge branch 'restore_bug' of https://github.com/LogoFX/logofx-client-mvvm-model into restore_bug

dmitrybublik committed:
Removed unnecessary test.
Fixed usings in DEBUG mode.

godrose committed:
#4 Removed usage of NotEditable Attribute

dmitrybublik committed:
Fixed snapshot by interface.

godrose committed:
#1 #4 removed obsolete not editable attribute

dmitrybublik committed:
Fixed usage of Bcl objects properties in EditableModel;
Fixed recursive reference snapshot.

dmitrybublik committed:
Merge branch 'restore_bug' of https://github.com/LogoFX/logofx-client-mvvm-model into restore_bug

dmitrybublik committed:
Split code for NET Standard 1.3 and 2.0

dmitrybublik committed:
Changed properties snapshot by fields snapshot.

godrose committed:
Minor

godrose committed:
Styling fixes

dmitrybublik committed:
Fixed ComplexSnapshot.

dmitrybublik committed:
Merge branch 'restore_bug' of https://github.com/LogoFX/logofx-client-mvvm-model into restore_bug

dmitrybublik committed:
Reverted output path

dmitrybublik committed:
Fixed Snapshots for list and arrays.

godrose reviewed (changes requested):
- [x] Remove EditableListAttribute from NETStandard 2.0
- [x] Remove NotEditableAttribute from the disk
- [x] Revert target framework change in the NETStandard13.Tests
- [x] BCLObjectInEditableModelTests - split into smaller tests; change names
- [x] CompositeEditableModel - Put Preprocessor directive next to all usages of EditableListAttribute
- [x] ConfigurationModelTests - change test name to the pattern Scenario_Condition_ExpectedResult
- [ ] ~~Add test for SelfEditableModel~~
- [ ] ~~Add test for TreeNodeEditableModel~~
- [x] Modify EditableModel.Snapshot files organization
- [ ] Bonus - add unit tests for Complex Snapshot
- [x] Remove EditablePropertyProxyAttribute
- [x] Add external errors to snapshot + tests
- [ ] ~~Bonus - Extract bcl tokens to external config file~~
- [x] Remove dependency on Newtonsoft.Json

dmitrybublik committed:
Removed EditableListAttribute from NETStandard 2.0;
Removed NotEditableAttribute from the disk;
Removed EditablePropertyProxyAttribute
Added external errors to snapshot

dmitrybublik committed:
Reverted target framework change in the NETStandard13.Tests

dmitrybublik committed:
CompositeEditableModel - Put Preprocessor directive next to all usages of EditableListAttribute

dmitrybublik committed:
Modified EditableModel.Snapshot files organization

godrose committed:
Updated FluentAssertions package

godrose committed:
Minor

dmitrybublik committed:
Fixed BCLObjectInEditableModelTests names.

dmitrybublik committed:
Added Extrenal Errors tests.

godrose committed:
Tests names changes

dmitrybublik committed:
Fixed SO bug in snapshot for Editable models with reference to itself.
Added tests.

dmitrybublik committed:
Merge branch 'restore_bug' of https://github.com/LogoFX/logofx-client-mvvm-model into restore_bug

dmitrybublik committed:
Removed dependency on Newtonsoft.Json
